warning: "V8_DISABLE_DEPRECATIONS" is not defined [-Wundef]

I get the compiler warning:

```
node_modules/nan/nan.h:39:27: warning: "V8_DISABLE_DEPRECATIONS" is not defined [-Wundef]
```

You should use `!defined(V8_DISABLE_DEPRECATIONS)` instead of just `!V8_DISABLE_DEPRECATIONS` at https://github.com/rvagg/nan/blob/master/nan.h#L39 and https://github.com/rvagg/nan/blob/master/nan.h#L41

--- a/nan.h
+++ b/nan.h
@@ -28,21 +28,25 @@
 #include <limits.h>
 #include <string>
 
-#if defined(__GNUC__) && !defined(DEBUG)
+#define notset(x) (!defined(x) || !x)
+
+#if defined(__GNUC__) && notset(DEBUG)
 # define NAN_INLINE inline __attribute__((always_inline))
-#elif defined(_MSC_VER) && !defined(DEBUG)
+#elif defined(_MSC_VER) && notset(DEBUG)
 # define NAN_INLINE __forceinline
 #else
 # define NAN_INLINE inline
 #endif
 
-#if defined(__GNUC__) && !V8_DISABLE_DEPRECATIONS
+#if defined(__GNUC__) && notset(V8_DISABLE_DEPRECATIONS)
 # define NAN_DEPRECATED __attribute__((deprecated))
-#elif defined(_MSC_VER) && !V8_DISABLE_DEPRECATIONS
+#elif defined(_MSC_VER) && notset(V8_DISABLE_DEPRECATIONS)
 # define NAN_DEPRECATED __declspec(deprecated)
 #else
 # define NAN_DEPRECATED
 #endif
+
+#undef notset
 
 // some generic helpers
 

